pkg/k8s/labels.go

```diff
@@ -423,6 +435,9 @@ const (
 	// GatewayPortName is the name of the incoming port of the gateway
 	GatewayPortName = "incoming-port"
 
+	// ProbePortName is the name of the probe port of the gateway
```

Comment:
Is there context or discussion somewhere about this change that I can read?  I don't really love using port names as part of the API here (or for the incoming-port).  Looking for ports with specific names feels a bit more magical and implicit as opposed to annotations which feel more explicit to me.  However, I don't feel strongly about this.

Reply:
Do we incur any code complexity due to using port names? It seems like a fine approach to me. Annotations have the downside of having to keep the two fields in-sync if you want to change it.

I might be inclined to change the port names to `linkerd-gateway` and `linkerd-gateway-probe` so it's clearer that we are in control of them and it's not arbitrary.  

I would definitely prefer that we _not_ include the `-port` suffix in these port names. We know their ports. What else could they be?

Reply:
Code complexity is not an issue here.  It's more that I tend to assume that names are not important (so long as they match up where they need to), but in this case these specific names have a special meaning which feels a little bit surprising to me.  But, again, not something I feel super strongly about.

How do you want to proceed from here, @olix0r?  I can merge this PR and then prepare a follow-up which renames the ports to `linkerd-gateway` and `linkerd-gateway-probe` if that sounds good.

Reply:
@adleong sgtm